### PR TITLE
Fix Build. Add Travis CI support

### PR DESCRIPTION
**Pull request by iMax-pp**
- Fix build
- Update README
- Add Travis CI support (e.g. https://travis-ci.org/iMax-pp/intools-engine)

### CODE REVIEW

File: intools.types.go [hunk 1]
```diff
@@ -9,7 +9,7 @@ import (
 )
 
 type IntoolsEngine struct {
-	DockerClient dockerclient.Client
+	DockerClient dockerclient.DockerClient
```
Comment (fsamin) [inline intools.types.go]: Why did you change the interface by the type ? Keeping the interface `dockerclient.Client` will be more reliable to mock Docker for unit tests

File: main.cli.app.go [hunk 2]
```diff
@@ -75,7 +75,7 @@ func runAction(c *cli.Context) {
 	cmd = cmd[4:]
 	Debug.Println("Launching " + image + " " + strings.Join(cmd, " "))
 	Warning.Printf("In command line, connector schedule is not available")
-	Intools = &IntoolsEngine{dockerClient, host, *redisClient, nil}
+	Intools = &IntoolsEngine{*dockerClient, host, *redisClient, nil}
```
Comment (fsamin) [inline main.cli.app.go]: needed when you use a type, not with an interface

File: main.daemon.go [hunk 3]
```diff
@@ -23,7 +23,7 @@ func newDaemon(port int, debug bool, dockerClient *dockerclient.DockerClient, do
 	}
 	engine := gin.Default()
 	cron := cron.New()
-	Intools = &IntoolsEngine{dockerClient, dockerHost, *redisClient, cron}
+	Intools = &IntoolsEngine{*dockerClient, dockerHost, *redisClient, cron}
```
Comment (fsamin) [inline main.daemon.go]: and so on...

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    - 1.4
+    - tip
+
+install:
+    - go get -u "github.com/codegangsta/cli"
+    - go get -u "github.com/gin-gonic/gin"
+    - go get -u "github.com/robfig/cron"
+    - go get -u "github.com/samalba/dockerclient"
+    - go get -u "gopkg.in/redis.v3"
+    - go build .
```
Comment (fsamin) [inline .travis.yml]: remove the dot `.`please, it will build the entire package

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+language: go
+
+go:
+    - 1.4
```
Comment (fsamin) [inline .travis.yml]: i run it with 1.4.1

File: intools.types.go (hunk 1 above)
Comment (iMax-pp) [inline intools.types.go]: It might be a problem with the `dockerclient` dependency, but on my machine I have the following error, when using `dockerclient.Client`:

```
dockerclient.DockerClient does not implement dockerclient.Client (ContainerChanges method has pointer receiver)
```

File: .travis.yml (hunk 5 above)
Comment (iMax-pp) [inline .travis.yml]: 1.4.2 on my side. Latest digit is a bug fix version, so 1.4 should suffice.

File: main.cli.app.go (hunk 2 above)
Comment (iMax-pp) [inline main.cli.app.go]: Problem related to previous error on `dockerclient.Client`.

File: .travis.yml (hunk 4 above)
Comment (iMax-pp) [inline .travis.yml]: Replaced with `go build -v ./...`.

**Comment by iMax-pp** [comment]
Issue opened on dockerclient repo: https://github.com/samalba/dockerclient/issues/134
And pull request: https://github.com/samalba/dockerclient/pull/135

File: intools.types.go (hunk 1 above)
Comment (fsamin) [inline intools.types.go]: Well done !

**Comment by iMax-pp** [comment]
Once the issue is fixed on `dockerclient` repository, I will change back `IntoolsEngine.DockerClient` type.

**Comment by fsamin** [comment]
yep, issue #2
